pkg/plugins/pfs/pfs.go

```diff
@@ -205,7 +205,13 @@ func ParsePluginFS(fsys fs.FS, rt *thema.Runtime) (ParsedPlugin, error) {
 	})
 	bi.Files = append(bi.Files, f)
 
-	gpi := ctx.BuildInstance(bi).Unify(gpv)
+	gpi := ctx.BuildInstance(bi)
+	// Temporary hack while we figure out what in the elasticsearch lineage turns
+	// this into an endless loop in thema, and why unifying twice is anything other
+	// than a total no-op.
+	if pp.Properties.Id != "elasticsearch" {
+		gpi = gpi.Unify(gpv)
+	}
```

Comment:
This is quite baffling.

This unification - which should be a redundant no-op for core plugins when they have [this line](https://github.com/grafana/grafana/pull/68034/files#diff-3be98bfef812f58d55f118e27326f4130a284fd285d2a192a7435204caef2de1R23) - causes the CUE evaluator to hang. And _only_ for the elasticsearch composable kind.

i could almost understand this happening for Cloudwatch, which contains a recursive type. I wonder if there's some bizarre incidental shadowing problem going on. Either way, this deserves further investigation - but we can do it in a follow-up.